src/macaron/slsa_analyzer/provenance/intoto/v01/__init__.py

```diff
@@ -161,7 +186,9 @@ def is_valid_digest_set(digest: dict[str, JsonType]) -> TypeGuard[dict[str, str]
         ``True`` if the digest set is valid according to the spec, in which case its type
         is narrowed to a ``dict[str, str]``; ``False`` otherwise.
     """
-    for value in digest.values():
-        if not isinstance(value, str):
+    for key in digest:
+        if key not in VALID_ALGORITHMS:
+            return False
```

Comment:
Sorry for missing this earlier.

According to the [Guidelines section on the Digest Set page](https://github.com/in-toto/attestation/blob/main/spec/v1/digest_set.md#guidelines):

> Consumers MUST only accept algorithms that they consider secure and MUST ignore unrecognized or unaccepted algorithms.

Therefore, I don't think we should restrict the digest set to only a set of known algorithms.

For Witness provenances, they also have `"gitoid:sha1"` and `"gitoid:sha256"` as keys in each subject's digest set, although these are not mentioned on the Digest Set page in the Supported Algorithms section. This valid algorithm restriction will cause Macaron to error while validating a Witness provenance payload.